tests/testChecker.ml

```diff
@@ -197,6 +197,250 @@ let suite =
+       let _, checker = Checker.entrypoint_mint_kit (checker, (Ligo.nat_from_literal "0n", (kit_of_mukit (Ligo.nat_from_literal "10_000_000n")))) in
+
+       Ligo.Tezos.new_transaction ~seconds_passed:1 ~blocks_passed:1 ~sender:alice_addr ~amount:(Ligo.tez_from_literal "0mutez");
+       let ops, _ = Checker.entrypoint_add_liquidity
+           (checker,
+            (* Note: all values here were arbitrarily chosen based on the amount of kit we mint above *)
+            ( ctez_of_muctez (Ligo.nat_from_literal "5_000_000n")
+            , kit_of_mukit (Ligo.nat_from_literal "5_000_000n")
+            , lqt_of_denomination (Ligo.nat_from_literal "5_000_000n")
+            , Ligo.timestamp_from_seconds_literal 999
+            )
+           ) in
+
+       let expected_ops = [
+         (LigoOp.Tezos.fa12_transfer_transaction
+            Fa12Interface.(
+              {address_to=checker_address;
+               address_from=alice_addr;
+               value=(Ligo.nat_from_literal "5_000_000n")}
+            )
+            (Ligo.tez_from_literal "0mutez")
+            (Option.get (LigoOp.Tezos.get_entrypoint_opt "%transfer" checker.external_contracts.ctez))
+         );
+       ] in
+       assert_operation_list_equal ~expected:expected_ops ~real:ops
+    );
+
+    ("entrypoint_burn_kit - emits expected operations" >::
+     fun _ ->
+       Ligo.Tezos.reset ();
+       let checker = empty_checker in
+       (* Create a burrow and mint some kit *)
+       Ligo.Tezos.new_transaction ~seconds_passed:10 ~blocks_passed:1 ~sender:alice_addr ~amount:(Ligo.tez_from_literal "100_000_000mutez");
+       let _, checker = Checker.entrypoint_create_burrow (checker, (Ligo.nat_from_literal "0n", None)) in
+       Ligo.Tezos.new_transaction ~seconds_passed:10 ~blocks_passed:1 ~sender:alice_addr ~amount:(Ligo.tez_from_literal "0mutez");
+       let _, checker = Checker.entrypoint_mint_kit (checker, (Ligo.nat_from_literal "0n", (kit_of_mukit (Ligo.nat_from_literal "10_000_000n")))) in
+       (* Then burn the kit *)
+       Ligo.Tezos.new_transaction ~seconds_passed:10 ~blocks_passed:1 ~sender:alice_addr ~amount:(Ligo.tez_from_literal "0mutez");
+       let ops, _ = Checker.entrypoint_burn_kit (checker, (Ligo.nat_from_literal "0n", (kit_of_mukit (Ligo.nat_from_literal "10_000_000n")))) in
+       assert_operation_list_equal ~expected:[] ~real:ops
+    );
+
+    ("entrypoint_create_burrow - emits expected operations" >::
+     fun _ ->
+       Ligo.Tezos.reset ();
+       Ligo.Tezos.new_transaction ~seconds_passed:10 ~blocks_passed:1 ~sender:alice_addr ~amount:(Ligo.tez_from_literal "100_000_000mutez");
+       let ops, _ = Checker.entrypoint_create_burrow (empty_checker, (Ligo.nat_from_literal "0n", None)) in
+
+       match ops with
+       (* Note: it's not really possible to check the first parameter of the contract here which is the
+        * function which defines the contract's logic.
+       *)
```

Comment:
Ya, it's not easy! This might be something that we can test at a higher level as well (e.g. if we end up adding another test layer using something like the mockup client).

Reply:
Yeah, that's a possibility. However, my hope is that adopting @utdemir's [idea](https://github.com/tezos-checker/checker/issues/213#issuecomment-887247912) will make the burrow contracts obsolete, effectively delegating the problem. Still have to iron out the details of this approach but I expect it to work.